shims/common/src/main/scala/org/apache/gluten/execution/BatchCarrierRowBase.scala

```diff
@@ -35,7 +34,7 @@ import org.apache.spark.unsafe.types.{CalendarInterval, UTF8String}
  * PlaceholderRows, followed by one TerminalRow that actually wraps that columnar batch. The total
  * number of PlaceholderRows + the TerminalRow equates to the size of the original columnar batch.
  */
-sealed abstract class BatchCarrierRow extends InternalRow {
+abstract class BatchCarrierRowBase extends InternalRow {
```

Comment:
We can avoid shiming this class by making `BatchCarrierRow` extend a mixin trait. I'll help update the PR for that.